lib/db.c

```diff
@@ -6075,7 +6075,13 @@ grn_table_setoperation_or(grn_ctx *ctx,
                                               key_src_size,
                                               &value_dest,
                                               &added);
-      if (id_dest != GRN_ID_NIL) {
+      if (id_dest == GRN_ID_NIL) {
+        if (ctx->rc == GRN_SUCCESS) {
+          continue;
+        } else {
+          break;
+        }
+      } else {
```

Comment:
```suggestion
      }

```

Could you also remove this `else` and unindent the current `else` block contents?